### DISCUSSION TITLE
[NOT A BUG] [Help needed] How to draw a building layout over the map

### DISCUSSION THREAD

**Discussion by aravi365** [react-native-maps/react-native-maps#4755] [Q&A]
### Summary

I am creating a booking app which allows user to select slots in a building and so i would like to draw a building layout in a particular building or coordinate and then let user to select slots from the layout. How to implement the same in a way that i can draw over maps - a layout of the building with clickable or selectable slots.
Here is the rought image of my expected implementation:
<img width="440" alt="Screenshot 2023-06-02 at 6 59 39 PM" src="https://github.com/react-native-maps/react-native-maps/assets/15327129/dbcc4cf9-87da-4d28-8a06-a0088aaaa421">

### Reproducible sample code

```TSX
<MapView
          provider={PROVIDER_GOOGLE} // remove if not using Google Maps
          style={styles.map}
          initialRegion={{
            latitude: 23.92374455995645,
            longitude: 47.26707199897953,
            latitudeDelta: 0.005,
            longitudeDelta: 0.0121,
          }}>
          {data.map((marker, index) => {
            return (
              <Marker
                key={index}
                coordinate={{latitude: marker.lat, longitude: marker.lng}}>
              </Marker>
            );
          })}
```

### Steps to reproduce

Need help on implementing the above functionality. Can it be implemented using Overlay and/or using Polyline?

### Expected result

A building Layout over the map with clickable areas

### Actual result

Not able to implement

### React Native Maps Version

1.7.1

### What platforms are you seeing the problem on?

Android, iOS (Apple Maps), iOS (Google Maps)

### React Native Version

0.71.8

### What version of Expo are you using?

Not using Expo

### Device(s)

All devices

**Comment by stjasink** [comment]
For the building outlines, just use polygons.  For the "slots" inside the buildings, use markers with custom shapes: https://github.com/react-native-maps/react-native-maps/blob/master/docs/marker.md#children-components
